txn-recovery-marker-jdbc-hibernate5/src/main/java/io/narayana/openshift/txrecovery/ApplicationRecoveryPod.java

```diff
@@ -0,0 +1,92 @@
+package io.narayana.openshift.txrecovery;
+
+import java.io.Serializable;
+
+import javax.persistence.Embeddable;
+import javax.persistence.EmbeddedId;
+import javax.persistence.Entity;
+import javax.persistence.Table;
+
+/**
+ * DTO for gathering name of application pod and
+ * the corresponding recovery pod name which the recovery pod works at.
+ */
+@Entity
+@Table(name = ApplicationRecoveryPod.TABLE_NAME)
+public class ApplicationRecoveryPod {
+    public static final String TABLE_NAME = "RECOVERY_MARKER";
+
+    @EmbeddedId private RecoveryInProgressRecordId id;
+
+    public ApplicationRecoveryPod() {
+        // constructor needed by Hibernate
+    }
+
+    public ApplicationRecoveryPod(String applicationPodName, String recoveryPodName) {
+        this.id = new RecoveryInProgressRecordId();
+        id.applicationPodName = applicationPodName;
+        id.recoveryPodName = recoveryPodName;
+    }
+
+    ApplicationRecoveryPod setApplicationPodName(String applicationPodName) {
+        this.id.applicationPodName = applicationPodName;
+        return this;
+    }
+    ApplicationRecoveryPod setRecoveryPodName(String recoveryPodName) {
+        this.id.recoveryPodName = recoveryPodName;
+        return this;
+    }
+
+    public String getApplicationPodName() {
+        return this.id.applicationPodName;
+    }
+    public String getRecoveryPodName() {
+        return this.id.recoveryPodName;
+    }
+
+    @Override
+    public String toString() {
+        return String.format("app pod name: %s, recovery pod name: %s",
+            getApplicationPodName(), getRecoveryPodName());
+    }
+}
+
+@Embeddable
```

Comment:
I haven't seen this before, it looks a bit wrongto define this class inside of ApplicationRecoveryPod.java - is it intentional?

Reply:
agree, that's probably not a good practice to define class inside of the class when there is no special intention for it ;)